examples/basic/evm-example/src/components/SwapContent/SwapContent.tsx

```diff
@@ -0,0 +1,599 @@
+      setError(`Could not get to wallet address.`)
+      return
+    }
+
+    const from: Asset = {
+      blockchain: fromToken.blockchain as string,
+      symbol: fromToken.symbol as string,
+      address: fromToken.address as string,
+    }
+    const to: Asset = {
+      blockchain: toToken.blockchain,
+      symbol: toToken.symbol as string,
+      address: toToken.address as string,
+    }
+    const amount: string = new BigNumber(inputAmount)
+      .shiftedBy(fromToken?.decimals as number)
+      .toString()
+
+    // const sampleImMessage = ethers.utils.defaultAbiCoder.encode(
+    //   ['(address,address)'],
+    //   [
+    //     [
+    //       fromToken.address || '0x0000000000000000000000000000000000000000',
+    //       userAddress,
+    //     ],
+    //   ]
+    // )
+
+    let request: QuoteRequest = {
+      amount,
+      from,
+      to,
+      swapperGroups: disabledLiquiditySources,
+      swappersGroupsExclude: true,
+    }
+    if (testMessagePassing) {
+      request = {
+        ...request,
+        messagingProtocols: selectedProtocols,
+        sourceContract: '???', // TODO
+        destinationContract: '???', // TODO
+      }
+    }
+
+    setLoadingSwap(true)
+
+    try {
+      const quote = await sdk.quote(request)
+      setQuote(quote)
+      console.log({ quoteResponse: quote })
+
+      if (
+        !quote ||
+        !quote?.route ||
+        quote.resultType !== RoutingResultType.OK
+      ) {
+        setError(
+          `Invalid quote response: ${quote.resultType}, please try again.`
+        )
+        setLoadingSwap(false)
+      } else {
+        await executeRoute(
+          from,
+          to,
+          fromAddress,
+          toAddress,
+          amount
+          // sampleImMessage
+        )
+      }
+    } catch (error) {
+      setError(`Error requesting quote: ${error}`)
+      setLoadingSwap(false)
+    }
+  }
+
+  const executeRoute = async (
+    from: Asset,
+    to: Asset,
+    fromAddress: string,
+    toAddress: string,
+    inputAmount: string
+    // imMessage?: string
+  ) => {
+    if (!fromToken || !toToken || !fromChain || !toChain) return
+
+    const signer = getSigners(
+      getWalletByChainType(fromChain?.type as TransactionType)
+    ).getSigner(fromChain?.type as TransactionType)
+
+    let swap: SwapResponse | null = null
+
+    try {
+      let swapRequest: SwapRequest = {
+        from,
+        to,
+        amount: inputAmount,
+        fromAddress: fromAddress,
+        toAddress: toAddress,
+        disableEstimate: false,
+        referrerAddress: null,
+        referrerFee: null,
+        slippage: '1.0',
+        swapperGroups: disabledLiquiditySources,
+        swappersGroupsExclude: true,
+      }
+      if (testMessagePassing) {
+        swapRequest = {
+          ...swapRequest,
+          messagingProtocols: selectedProtocols,
+          sourceContract: '???', // TODO
+          destinationContract: '???', // TODO
+          // imMessage,
```

Comment:
We should pass it when test message passing is enabled in ui by user and tx type is EVM.

Reply:
fixed